src/Item.js

```diff
@@ -1,33 +1,72 @@
 import React, { useState, useEffect } from 'react';
 import { db } from './lib/firebase';
+import estimates from './lib/estimates';
+import { differenceInDays, fromUnixTime } from 'date-fns';
 
-function Item(props) {
-  const { purchaseDates, userToken, itemName, id } = props;
+function Item({ userToken, item }) {
+  const { itemName, id, purchaseDates, purchaseEstimates } = item;
```

Comment:
Now that we have a `purchaseEstimates` field, there are several lists and grocery items previously added that do not have that field. That means if a grocery item is checked which doesn't have the field, it throws an error since `purchaseEstimates` is undefined and we are trying to get the length of it. This can be fixed easily with a default value:

https://user-images.githubusercontent.com/1530227/117512714-02d43d00-af45-11eb-87f9-207f16c902be.mov


```suggestion
  const { itemName, id, purchaseDates, purchaseEstimates  = [] } = item;
```

Reply:
Oh that's a great point! Thanks @luisaugusto